### PR TITLE
KT-34218 "Merge 'else if' intention"

### PR DESCRIPTION
**Pull request by DmiitriyJarosh**
https://youtrack.jetbrains.com/issue/KT-34218

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.psi.*
+
+
+class MergeElseIfIntention : SelfTargetingIntention<KtIfExpression>(KtIfExpression::class.java, "Merge 'else if'") {
+    override fun isApplicableTo(element: KtIfExpression, caretOffset: Int): Boolean {
+        val elseBody = element.`else` ?: return false
+        val nestedIf = elseBody.nestedIf() ?: return false
+        if (nestedIf.`else` != null) {
```
Comment (goodwinnk) [inline idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt]: Return can be lifted up.

File: idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        return true
+    }
+
+    override fun applyTo(element: KtIfExpression, editor: Editor?) {
+        applyTo(element)
+    }
+
+    companion object {
+        fun applyTo(element: KtIfExpression): Int {
```
Comment (goodwinnk) [inline idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt]: Can function be inlined? Also what is the purpose of return value?

File: idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            val nestedIf = element.`else`?.nestedIf() ?: return -1
+            val condition = nestedIf.condition ?: return -1
+            val nestedBody = nestedIf.then ?: return -1
+
+            val factory = KtPsiFactory(element)
+            val newBody = element.`else`?.replace(
+                factory.createExpressionByPattern("if ($0) $1", condition, nestedBody)
+            ) ?: return -1
+            return newBody.textRange!!.startOffset
```
Comment (goodwinnk) [inline idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt]: Please avoid using `!!`. If you sure that some nullable expression can't be null, be explicit in reasoning and use `?: error("An explanation why value can't be null")`.

File: idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt (hunk 2 above)
Comment (DmiitriyJarosh) [reply idea/src/org/jetbrains/kotlin/idea/intentions/MergeElseIfIntention.kt]: Return value was unnecessary. Fixed it.

**Comment by goodwinnk** [comment]
Thank you for the pull request and congrats on becoming a Kotlin contributor! I have merged your commit manually as https://github.com/JetBrains/kotlin/commit/8c4dd1f331f304ccc9bb5d61137dcc4de67abb8f.

I did some modifications in commit that might be important to know for future contributions:
- Please add ticket number to commit message
- Don't forget to commit generated test files
- Avoid warnings in the new code (there was a one because of using inline function)

**Pull request closed without merging**